model/version_activation.go

```diff
@@ -32,7 +39,83 @@ func DoProjectActivation(ctx context.Context, id string, ts time.Time) (bool, er
 	}
 
 	return activated, nil
+}
+
+func activateEveryRecentMainlineCommitForProject(ctx context.Context, projectRef *ProjectRef, ts time.Time) (bool, error) {
```

Comment:
although this is a private function and doesn't _need_ a function comment, I think it would be good here (or in DoProjectActivation) to explain the different behavior.